wom/src/main/scala/wom/values/WomFile.scala

```diff
@@ -1,48 +1,69 @@
 package wom.values
 
-import wom.types.{WomFileType, WomType}
+import wom.types.{WomFileType, WomGlobFileType, WomSingleFileType, WomType}
 
 import scala.util.{Success, Try}
 
-object WomFile {
-  def appendPathsWithSlashSeparators(path1: String, path2: String) = {
-    if (path1.endsWith("/") || path2.startsWith("/")) path1 + path2
-    else path1 + "/" + path2
-  }
-
-  def apply(value: String, isGlob: Boolean = false): WomFile = if (isGlob) WomGlobFile(value) else WomSingleFile(value)
-}
-
 sealed trait WomFile extends WomPrimitive {
   val value: String
-  val womType: WomType = WomFileType
 
-  def isGlob: Boolean = this match {
-    case _: WomGlobFile => true
-    case _ => false
-  }
-
-  override def add(rhs: WomValue): Try[WomValue] = rhs match {
-    case r: WomString => Success(WomFile(value + r.value))
-    case r: WomOptionalValue => evaluateIfDefined("+", r, add)
-    case _ => invalid(s"$value + $rhs")
-  }
+  override def valueString = value
 
+  // TODO: WOM: WOMFILE: When WDL supports directories this check may need more work (refactoring to subclasses?)
   override def equals(rhs: WomValue): Try[WomBoolean] = rhs match {
-    case r: WomFile => Success(WomBoolean(value.equals(r.value) && isGlob.equals(r.isGlob)))
-    case r: WomString => Success(WomBoolean(value.toString.equals(r.value.toString) && !isGlob))
+    case r: WomFile => Success(WomBoolean(value.equals(r.value) && womType.equals(r.womType)))
+    case r: WomString => Success(WomBoolean(value.toString.equals(r.value.toString) && womType == WomSingleFileType))
     case r: WomOptionalValue => evaluateIfDefined("==", r, equals)
     case _ => invalid(s"$value == $rhs")
   }
+}
+
+object WomFile {
+  def apply(fileType: WomFileType, value: String) = {
+    fileType match {
+      case WomSingleFileType => WomSingleFile(value)
+      case WomGlobFileType => WomGlobFile(value)
+    }
+  }
+
+  def mapFile(womFile: WomFile)(f: String => String): WomFile = {
+    womFile match {
+      case womSingleFile: WomSingleFile => mapSingleFile(womSingleFile)(f)
+      case womGlobFile: WomGlobFile => mapGlobFile(womGlobFile)(f)
+    }
+  }
 
-  override def valueString = value.toString
+  def mapSingleFile(womFile: WomSingleFile)(f: String => String): WomSingleFile = {
+    WomSingleFile(f(womFile.value))
+  }
+
+  def mapGlobFile(womFile: WomGlobFile)(f: String => String): WomGlobFile = {
+    womFile match {
+      case WomGlobFile(value) => WomGlobFile(f(value))
```

Comment:
why the pattern match on the known type here but just the member access above in `mapSingleFile`?

Reply:
Uhm...... this was a test for reviewers. You passed!